tests/func/test_remote.py

```diff
@@ -154,6 +154,8 @@ def test_dir_hash_should_be_key_order_agnostic(tmp_dir, dvc):
         _, _, obj = stage(dvc.odb.local, path, dvc.odb.local.fs, "md5")
         hash1 = obj.hash_info
 
+    remove(dvc.odb.local.cache_dir)
+
```

Comment:
Could you explain why, please?

Reply:
When we call `stage()`, we also create a raw dir hash for the tree. When calling `stage` the second time we load the tree from the raw dir hash. Since `HashInfo` values are set to dummy values, `hash_to_path` (which **expects** hashes to be longer than 3 characters) returns a dir (`memory://dvc-staging/3701baaa21c459498c4c21986d6a85b80d016815f63942d17fb650599e4d1558/1/`). This results in a failure when adding the tree to staging.

The solution was to just avoid loading the raw dir hash by removing it.
To make things clearer I could just remove the raw dir hash and add a note:
```diff
-    remove(dvc.odb.local.cache_dir)
+    # remove the raw dir hash so that the next `stage` call builds the tree instead of loading it from the raw dir hash
+    odb.fs.remove(odb.hash_to_path("55be181145ff2aa155e719083a5ffb64"))

The test would also pass by replacing the hashes when creating the dummy tree:
```diff
-        [{"relpath": "1", "md5": "1"}, {"relpath": "2", "md5": "2"}]
+        [{"relpath": "1", "md5": "123"}, {"relpath": "2", "md5": "234"}]
```
so that `hash_to_path()` would work correctly, but this would essentially make this test useless as the tree would just be loaded from the raw dir hash.

We could also add an assert in hash to path in order to avoid similar issues in the future:

```diff
    def hash_to_path(self, hash_):
+       assert len(hash_) > 2
        return self.fs.path.join(self.fs_path, hash_[0:2], hash_[2:])
```